### PR TITLE
feat(editor): 장면 진입 효과에 단서 지급 추가

### LINKED ISSUE #504: 일반 장면의 진입 효과를 정보 공개와 단서 지급 중심으로 재설계
## 목표

일반 phase 노드의 `InformationDeliveryPanel`을 `장면 진입 효과` 개념으로 재설계한다. 읽기 대사 배치는 제거하고, 특정 대상에게 정보 공개와 단서 지급을 설정할 수 있게 한다.

## 배경

사용자 피드백 기준으로 일반 장면에 들어왔을 때 발생할 수 있는 것은 다음과 같다.

- 특정 캐릭터 또는 전체 대상에게 정보가 공개된다.
- 특정 캐릭터 또는 전체 대상에게 단서가 지급될 수 있다.
- 읽기 대사는 여기서 다루지 않는다. 읽기 대사는 `스토리 진행` 노드 전용이다.

현재 문제:

- 상단에 `전체 대상 추가 / 캐릭터별 대상 추가`가 있는데 아래에 다시 `캐릭터 검색`이 있어 대상 선택 UX가 중복된다.
- `읽기 대사 배치`가 일반 장면의 정보 공개 카드 안에 섞여 있다.
- 단서 지급 진입점이 없다.

## 작업 범위

- `InformationDeliveryPanel`을 `SceneEntryEffectPanel` 또는 동등한 이름의 컴포넌트로 개편한다.
- 기존 `읽기 대사 배치` UI 제거.
- `StoryInfoGrantPanel` 유지/분리.
  - 대상에게 공개할 스토리 정보를 선택한다.
- `ClueGrantPanel` 신설.
  - 대상에게 지급할 단서를 선택한다.
  - 기존 `useEditorClues` 또는 관련 clue API를 사용한다.
- `RecipientTargetPanel` 신설 또는 분리.
  - `전체 대상`과 `캐릭터별 대상` 선택 흐름을 하나로 통합한다.
  - 상단 버튼과 하단 캐릭터 검색이 중복으로 보이지 않게 한다.
- adapter 이름과 ViewModel 재정리.
  - 기존 `InformationDeliveryViewModel`에서 `readingSectionIds` 제거.
  - 신규 ViewModel 예시:

```ts
interface SceneEntryEffectViewModel {
  id: string;
  recipientType: "all_players" | "character";
  characterId?: string;
  storyInfoIds: string[];
  clueIds: string[];
}
```

- 저장 shape는 기존 `deliver_information` action을 확장할지, 신규 action을 둘지 코드와 runtime을 확인해 결정한다.

### 병렬 가능 작업

- `mmp-frontend-editor-reviewer`: 대상 선택, 정보 공개, 단서 지급 UX가 제작자에게 자연스러운지 리뷰한다.
- `mmp-backend-engine-reviewer`: 단서 지급이 runtime에서 어떤 action/module로 해석되어야 하는지 확인한다. backend 변경 필요 여부를 판단한다.
- `mmp-test-coverage-reviewer`: adapter와 component test가 저장 payload와 edge case를 충분히 덮는지 리뷰한다.

### 파일/모듈 소유권

- 주 작업자 소유:
  - `apps/web/src/features/editor/components/design/InformationDeliveryPanel.tsx`
  - `apps/web/src/features/editor/components/design/InformationDeliveryPanelViews.tsx`
  - `apps/web/src/features/editor/components/design/SceneEntryEffectPanel.tsx` 신규 가능
  - `apps/web/src/features/editor/components/design/RecipientTargetPanel.tsx` 신규 가능
  - `apps/web/src/features/editor/components/design/StoryInfoGrantPanel.tsx` 신규 가능
  - `apps/web/src/features/editor/components/design/ClueGrantPanel.tsx` 신규 가능
  - `apps/web/src/features/editor/entities/shared/informationDeliveryAdapter.ts` 또는 신규 `sceneEntryEffectAdapter.ts`
- API/데이터 확인:
  - `apps/web/src/features/editor/api.ts`
  - `apps/web/src/features/editor/editorClueApi.ts`
  - `apps/server/internal/module/progression/information_delivery/**`
  - 단서 지급 runtime 관련 module/service 파일
- 테스트:
  - `apps/web/src/features/editor/components/design/__tests__/InformationDeliveryPanel.test.tsx`
  - 신규 `SceneEntryEffectPanel.test.tsx`
  - adapter 테스트

### 병렬 금지/주의 영역

- 단서 지급이 실제 runtime effect로 이어지는 계약은 frontend-only mock으로 끝내면 안 된다.
- backend action/migration이 필요하면 별도 PR 또는 명확한 backend lane으로 분리한다.
- #503의 읽기 대사 분리와 같은 파일을 만질 가능성이 높으므로 PR 순서를 조정한다.

### 취합 방식

각 subagent는 `발견 / 수행 / 판단 / 미해결` 형식으로 보고한다. 메인 Codex가 저장 계약, backend 필요 여부, PR 분할을 최종 결정한다.

## 제외

- 스토리 진행 노드의 읽기 대사 전용 패널 구현. 이는 #503에서 처리한다.
- 단서 자체의 생성/편집 UI 변경.
- 정보 관리 탭 자체의 편집 기능 변경.
- 복잡한 조건식 기반 지급. MVP는 장면 진입 시 대상에게 지급하는 구조로 제한한다.

## 완료 조건

- 일반 장면 패널에서 읽기 대사 검색/배치가 제거된다.
- 대상 선택 UX가 중복되지 않는다.
- 정보 공개는 유지된다.
- 단서 지급 UI가 추가된다.
- 저장 payload와 runtime 해석 경계가 테스트 또는 follow-up으로 추적된다.

## Coverage Plan

- component test:
  - 전체 대상 effect 추가.
  - 캐릭터 대상 effect 추가.
  - 정보 선택/해제.
  - 단서 선택/해제.
  - 대상 미선택 또는 지급 항목 없음 상태 처리.
- adapter test:
  - 기존 `story_info_ids` 호환.
  - 신규 `clue_ids` 변환.
  - 중복 id 제거.
  - 빈 effect 제거.
- backend 변경이 포함되면:
  - action validation test.
  - runtime effect/idempotency test.
  - 대상 캐릭터 not found 또는 clue not found 실패 경로 test.

## 검증 계획

- `pnpm --filter @mmp/web test -- InformationDeliveryPanel SceneEntryEffectPanel sceneEntryEffectAdapter`
- `pnpm --filter @mmp/web typecheck`
- backend 변경 시 `cd apps/server && go test ./internal/module/... ./internal/domain/...`
- 브라우저에서 일반 phase 노드 선택 후 정보 공개와 단서 지급이 각각 독립적으로 보이는지 확인.

## PR 묶음 제안

프론트 UI/adapter만으로 가능한 범위는 하나의 PR로 처리한다. 단서 지급 runtime 계약 변경이 필요하면 backend PR을 별도로 분리하거나, 변경 범위가 작을 때만 같은 PR로 묶는다.

## 브레인스토밍 필요 여부

필요. 단서 지급이 기존 module/action으로 충분한지, 신규 action이 필요한지는 backend runtime 확인 후 결정해야 한다.

## 순서/의존성

#503 이후 진행을 권장한다. 읽기 대사 제거 경계가 먼저 정리되어야 일반 장면 진입 효과가 명확해진다.

### PR DESCRIPTION
**Pull request by sabyunrepo**
## 요약

- 일반 장면의 `InformationDeliveryPanel`을 `장면 진입 효과` 역할로 개편해 정보 공개와 단서 지급을 대상별로 함께 설정합니다.
- `GRANT_CLUE` phase action을 backend runtime에 추가하고 `clue_interaction` module에서 캐릭터/전체 대상 지급을 처리합니다.
- 일반 트리거 추가 메뉴에서는 `DELIVER_INFORMATION`/`GRANT_CLUE`가 중복 노출되지 않게 숨겼습니다.

Closes #504

## Coverage Plan

- backend engine/action:
  - `GRANT_CLUE` action 등록, legacy alias, implicit module 연결을 확인합니다.
  - `clue_interaction`의 캐릭터 대상 지급, 전체 대상 지급, 중복 실행 방지, 빈 clue validation을 테스트합니다.
- frontend adapter:
  - 기존 `DELIVER_INFORMATION`과 신규 `GRANT_CLUE` action을 하나의 `SceneEntryEffectViewModel`로 병합합니다.
  - 저장 시 정보 공개와 단서 지급을 별도 runtime action으로 쓰고, 빈 effect와 중복 id를 정리합니다.
- frontend component:
  - 캐릭터 대상 추가, 전체 대상 추가, 정보 선택/삭제, 단서 선택/삭제, empty/error 상태를 테스트합니다.
  - 일반 장면과 `story_progression` 노드의 패널 분리를 유지합니다.

## Local validation

- `cd apps/server && go test ./internal/engine ./internal/module/core` 성공
- `pnpm --dir apps/web exec vitest run src/features/editor/entities/shared/__tests__/actionAdapter.test.ts src/features/editor/entities/shared/__tests__/sceneEntryEffectAdapter.test.ts src/features/editor/components/design/__tests__/InformationDeliveryPanel.test.tsx src/features/editor/components/design/__tests__/PhaseNodePanelExtended.test.tsx` 성공: 4 files / 30 tests
- `pnpm --dir apps/web typecheck` 성공
- Browser smoke 성공:
  - 일반 장면에서 `장면 진입 효과` 표시 확인
  - `장면 시작 트리거` 추가 메뉴에 `정보 공개`/`단서 지급` 중복 노출 없음 확인
  - E2E fixture에 정보/단서 데이터가 없어 실제 선택 UI는 component test로 보강
- `scripts/mmp-local-ci.sh quick` 성공
  - marker: `0b9d43a1d3191dac34638f0268a37430048984d9`, status `success`

## 참고

- `GRANT_CLUE`는 frontend-only로 저장하면 runtime 효과가 없기 때문에 이번 PR에 작은 backend action/module 처리를 함께 포함했습니다.
- `vite build` 중 기존 editor API barrel re-export 순환 warning이 출력되지만 build는 성공했습니다. 이 PR에서 새로 만든 직접 원인은 아니며 기존 파일들도 같은 warning을 내고 있습니다.

## Summary by CodeRabbit

* **새로운 기능**
  * 런타임에서 특정 캐릭터 또는 모든 플레이어에게 단서를 지급하는 동작이 추가되어 지급 적용, 중복 방지, 타깃 해석 및 지급 이벤트 발행을 지원합니다.
  * 에디터에 단서 검색·선택 및 단서 토글을 포함한 “장면 진입 효과” UI와 옵션이 추가되어 단서 지정이 가능해졌습니다.

* **리팩터**
  * 정보 공개 흐름이 장면 진입 효과 모델로 통합되어 관련 어댑터·뷰모델이 정리되고 편집기 문구가 갱신되었습니다.

* **테스트**
  * 단서 지급 로직, 어댑터 변환, 편집기 UI를 검증하는 단위·통합 테스트가 대폭 추가·확장되었습니다.

### CODE REVIEW

File: apps/web/src/features/editor/entities/shared/sceneEntryEffectAdapter.ts [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+      : "character";
+  if (deliveryType !== effect.recipientType) return false;
+  if (effect.recipientType === "all_players") return true;
+  const deliveryCharacterId =
+    typeof delivery.target?.character_id === "string"
+      ? delivery.target.character_id
+      : typeof delivery.character_id === "string"
+        ? delivery.character_id
+        : undefined;
+  return Boolean(effect.characterId) && deliveryCharacterId === effect.characterId;
```
[the remark this answers is not in the document]
Comment (sabyunrepo) [reply apps/web/src/features/editor/entities/shared/sceneEntryEffectAdapter.ts]: Fixed and covered. Removed the id-first shortcut so legacy reading_section_ids are carried forward only when the recipient matches, and added a regression test for same effect id with a different character target. Local validation: focused sceneEntryEffectAdapter Vitest passed, and scripts/mmp-local-ci.sh quick passed on f0d5e71b.

**Comment by sabyunrepo** [comment]
Fixed in the touched test file and tracked the broader existing cleanup separately as #518. Added afterAll(() => vi.unstubAllGlobals()) to sceneEntryEffectAdapter.test.ts so the crypto stub does not leak across tests. Focused validation: pnpm --dir apps/web exec vitest run src/features/editor/entities/shared/__tests__/sceneEntryEffectAdapter.test.ts passed on 39dcb3b2.
